### DISCUSSION TITLE
Allow one to add words to the exclusion (.exc) file

### DISCUSSION THREAD

**Discussion by doolio** [minad/jinx#115] [Ideas]
Per [enchant/issues/342](https://github.com/AbiWord/enchant/issues/342).

**Comment by minad** [comment]
There doesn't seem to be an API to add words. I think this file is supposed to be edited manually. How do you imagine the UI to add undesired words? It doesn't seem to make much sense to add words from the jinx-correct UI since there we rather modify inclusion lists instead of exclusion lists.

**Reply by minad** [reply]
But good idea asking about these EXC files. I had wondered about these files myself 😆

**Comment by doolio** [comment]
> There doesn't seem to be an API to add words. 

Ah I see. I thought it was just of matter of adding to these options:

![image](https://github.com/minad/jinx/assets/13859009/1ad5bd27-7911-4af7-8d89-321770d2f375)

**Reply by minad** [reply]
Sure, but I question the usefulness of doing that, as I wrote above:

> It doesn't seem to make much sense to add words from the jinx-correct UI since there we rather modify inclusion lists instead of exclusion lists.

**Reply by doolio** [reply]
The existing UI allows us to add words considered incorrect by enchant to our personal dictionary so that enchant will no longer consider them incorrect. 

The exclusion file is for words considered correct by enchant but we consider incorrect so we add it to this file so that enchant will consider it incorrect. Now that I write that out I don't see the purpose of this file. Does it make sense to you?

Feel free to close this discussion.
